nautobot/core/api/serializers.py

```diff
@@ -137,13 +138,41 @@ def __init__(self, *args, **kwargs):
                 kwargs["view_name"] = get_route_for_model(kwargs["queryset"].model, "detail", api=True)
         super().__init__(*args, **kwargs)
 
+    @property
+    def _related_model(self):
+        """The model class that this field is referencing to."""
+        if self.queryset is not None:
+            return self.queryset.model
+        # Foreign key where the destination is referenced by string rather than by Python class
+        if getattr(self.parent.Meta.model, self.source, False):
+            return getattr(self.parent.Meta.model, self.source).field.model
+
+        logger.warning(
+            "Unable to determine model for related field %r; "
+            "ensure that either the field defines a 'queryset' or the Meta defines the related 'model'.",
+            self.field_name,
+        )
+        return None
```

Comment:
It seems like this warning might be redundant seeing as `RelatedField` has this:

https://github.com/encode/django-rest-framework/blob/15c613a9eb645c63102b9e894199bcf1c9bf4d65/rest_framework/relations.py#L107-L115